src/Components/Invite/Invite.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom'
+import Spinner from '../Spinner';
+import { auth } from '../Firebase';
+import server from '../Server';
+
+class Invite extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            user: null,
+            gameID: props.location.pathname.split('/')[2],
+            redirect: false
+        }
+
+        this.initAuthListener = this.initAuthListener.bind(this);
+    }
+
+    componentDidMount() {
+        this.initAuthListener();
+    }
+
+    initAuthListener(){
+        auth.onAuthStateChanged(function (user) {
```

Comment:
Somewhere in this class we should wait for this to actually do something. I went to invite without being signed in and was told "couldn't join game" before being redirected to `/join` and then joining the game successfully

Reply:
I think that's because we redirect without waiting for the user to sign in

Reply:
I've changed this to redirect to the login page if the user is not signed in.

However after signing in, the user is not automatically redirected to the `/invite` link that they requested - they have to type it again.

Reply:
I've removed this and made it wait until the user is logged in. (`join` is now called only from the user being logged in)